File: 179588880-main/sentimental-readability/readability.py

def count_letters(text):
    sum = 0
    for i in range(0, len(text)):
        if text[i].isalpha():
            sum += 1
    return sum


def count_words(text):
    sum = 1
    for i in range(0, len(text)):
        if text[i] == " ":
            sum += 1
    return sum


def count_sentences(text):
    sentence_endings = ".!?"  # Cümle bitişlerini belirleyen karakterler
    sum = 0
    for i in text:
        if i in sentence_endings:
            sum += 1
    return sum


def main():
    text = input("Lütfen metini giriniz:")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = (letters / words) * 100
    S = (sentences / words) * 100

    index = 0.0588 * L - 0.296 * S - 15.8

    if index < 1:
        print("Before Grade 1")
    elif (index >= 16):
        print("Grade 16+")
    else:
        print("Grade {}".format(round(index)))


if __name__ == "__main__":
    main()

File: 179588880-main/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Kullanıcıdan metin girişi
    string text = get_string("Text: ");

    // Mektup, kelime ve cümle sayılarını say
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // L ve S hesaplamaları, tamsayıların float'a çevrilmesi önemli
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;

    // Coleman-Liau index formülü
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Sonuçları yazdır
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", round(index)); // Yuvarlayarak tam sayı çıktısı
    }
}

// Harf sayma fonksiyonu
int count_letters(string text)
{
    int sum = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i])) // isalpha() fonksiyonu kullanarak harf olup olmadığını kontrol edin
        {
            sum++;
        }
    }
    return sum;
}

// Kelime sayma fonksiyonu
int count_words(string text)
{
    int sum = 1; // Kelime sayısı en az 1 olmalı
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ') // Boşluklara bakarak kelimeleri sayın
        {
            sum++;
        }
    }
    return sum;
}

// Cümle sayma fonksiyonu
int count_sentences(string text)
{
    int sum = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' ||
            text[i] == '?') // Noktalama işaretleriyle cümleleri sayın
        {
            sum++;
        }
    }
    return sum;
}
